module/zfs/vdev_raidz.c

```diff
@@ -2290,6 +2336,23 @@ vdev_raidz_child_done(zio_t *zio)
 	rc->rc_error = zio->io_error;
 	rc->rc_tried = 1;
 	rc->rc_skipped = 0;
+
+	/*
+	 * Process the disk io latency before it goes out of scope.
+	 *
+	 * A zio->io_delay value of zero means this IO was part of
+	 * an aggregation.
+	 */
+	if (zio->io_type == ZIO_TYPE_READ && zio->io_error == 0 &&
```

Comment:
What do you think about adding if `raidz_read_sit_out_secs != 0` to this check to skip calculating the ewma when this detection is disabled.  It might help a little.

Reply:
Good idea. Added
```@@ -2343,8 +2343,8 @@ vdev_raidz_child_done(zio_t *zio)
         * A zio->io_delay value of zero means this IO was part of
         * an aggregation.
         */
-       if (zio->io_type == ZIO_TYPE_READ && zio->io_error == 0 &&
-           zio->io_size > 0 && zio->io_delay != 0) {
+       if (raidz_read_sit_out_secs != 0 && zio->io_type == ZIO_TYPE_READ &&
+           zio->io_error == 0 && zio->io_size > 0 && zio->io_delay != 0) {
                vdev_t *vd = zio->io_vd;
                uint64_t previous_ewma = atomic_load_64(&vd->vdev_ewma_latency);
                if (previous_ewma == 0)
```